File: remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        length = 0
        for idx, val in enumerate(nums):
            if idx == 0 or val != nums[idx-1]:
                nums[length] = val
                length += 1
        
        return length
        

File: remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int length = 0;
        for (int i = 0 ; i < nums.length ; i++) {
            if (i == 0 || nums[i] != nums[i-1]) {
                nums[length] = nums[i];
                length++;
            }
        }
        return length;
        
        
    }
}

// Alternatively, where wp is a writer pointer and i is read pointer
 
class Solution {
    public int removeDuplicates(int[] nums) {
        int wp = 1;
        for (int i = 0; i < nums.length; i++) {
            if (i + 1 < nums.length) {
                nums[wp] = nums[i+1];
                wp++;
            }
        }
        return wp;
    }
}

